src/System.Memory/src/System/Buffers/BuffersExtensions.cs

```diff
@@ -115,6 +115,9 @@ public static void Write<T>(this IBufferWriter<T> writer, ReadOnlySpan<T> value)
         {
             Span<T> destination = writer.GetSpan();
 
+            if (destination.IsEmpty)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.destination);
```

Comment:
Is "destination" the right argument name to use here?  There's no "destination" in the public API.

Also, what about the case where we call WriteMultiSegment and its call to GetSpan returns an empty span?

Reply:
> Also, what about the case where we call WriteMultiSegment and its call to GetSpan returns an empty span?

+1, I don't think this change alone avoids the infinite loop.